docs/serverless-v2/03-03-lambda-processing-phases.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Lambda processing phases
+type: Details
+---
+
+From the moment you create or update a lambda (Function CR) until the time it reaches the final `Running` status phase, the lambda goes through three processing phases:
+
+1. `Initializing`
+2. `Building`
+3. `Deploying`
+
+The diagrams illustrate these three core phases of a lambda processing circle that the Function Controller handles. They also list all custom resources involved in this process and specify in which cases their update is required.
+
+>**NOTE:** Before you start reading, see the [Function CR](#custom-resource-function) document for the custom resource detailed definition, the list of all lambda status phases and reasons for their success or failure.
+
+## Initializing
+
+This initial phase starts when you create a Function CR with configuration specifying the lambda's setup. It ends with creating a ConfigMap and a TaskRun CR ready for building the lambda image.
+
+Updating an existing lambda involves an image rebuild only if any previous lambda processing phase failed (**phase: Failed**), or if you change lambda's configuration, so its body (**function**) or dependencies (**deps**). An update of lambda's environmental variables doesn't require image rebuild, and it only affects KService in the `Deploying` phase.
+
+> **NOTE:** Each time you update lambda's configuration, the Function Controller deletes all previous TaskRun CRs for the given lambda's **UID**.
+
+![Initializing stage](./assets/initializing.svg)
```

Comment:
When the rebuild is not required then it rather goes to `Deploying` phase

Reply:
Actually when we modify Function CR and the change does not required rebuild, we still need go to `Building` phase. The build may still be in progress (if the change was made during the build) and we need to wait it finish.

Reply:
@michal-hudy is that ok for you or maybe some clarification should be added to the docs?

Reply:
It is OK